cmd/topicctl/subcmd/apply.go

```diff
@@ -270,8 +316,30 @@ func applyTopic(
 			SleepLoopDuration:          applyConfig.sleepLoopDuration,
 			TopicConfig:                topicConfig,
 		}
+		topicChanges, err := cliRunner.ApplyTopic(ctx, applierConfig)
+		if err != nil {
+			// if one of the steps after updateSettings errors when updating a topic,
+			// we can be in a state where some (but not all) changes were applied
+			if topicChanges.NewChanges != nil || topicChanges.UpdateChanges != nil {
+				changesToBePrinted = unpackChanges(topicChanges, changesToBePrinted)
+				log.Error("Error detected while updating topic, the following changes were still made:")
+				partialChanges, err := printChanges(changesToBePrinted)
+				if err != nil {
+					log.Error("Error printing JSON changes data")
+					return err
+				}
+				log.Errorf("%#v", partialChanges)
+			}
+			return err
+		}
+		if topicChanges.NewChanges != nil || topicChanges.UpdateChanges != nil {
```

Comment:
When can this be totally empty ?

Reply:
I think they used to be totally empty when there were no changes on a topic, but I removed the if statement and nothing changed/all tests pass so I'll commit the removal